pkgs/development/tools/build-managers/xmake/default.nix

```diff
@@ -1,48 +1,30 @@
-{ lib
-, stdenv
-, fetchurl
-, pkg-config
-, lua
-, readline
-, ncurses
-, lz4
-, tbox
-, xmake-core-sv
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  CoreServices,
+  ...
 }:
-
 stdenv.mkDerivation rec {
   pname = "xmake";
-  version = "2.9.4";
-
-  src = fetchurl {
-    url = "https://github.com/xmake-io/xmake/releases/download/v${version}/xmake-v${version}.tar.gz";
-    hash = "sha256-deLd4r0qSKMymJuAGuZQd8RS1JH+xRep2yeoHIcTzcU=";
+  version = "v2.9.5";
+  src = fetchFromGitHub {
+    owner = "xmake-io";
+    repo = "xmake";
+    rev = version;
+    hash = "sha256-/hzYrdB+0up4o78zE1KkhQY0N1trnd9fcrp7s/spVfo=";
+    fetchSubmodules = true;
   };
 
-  nativeBuildInputs = [
-    pkg-config
-  ];
-
-  buildInputs = [
-    lua
-    lua.pkgs.cjson
-    readline
-    ncurses
-    lz4
-    tbox
-    xmake-core-sv
-  ];
+  patches = [ ./0001-Use-GNU-find-and-date-by-removing-unnecessary-host-c.patch ];
 
-  strictDeps = true;
+  buildInputs =
+    [
+    ]
+    ++ lib.optional stdenv.hostPlatform.isDarwin CoreServices;
 
-  configureFlags = [ "--external=y" ];
+  preConfigure = ''
+    patchShebangs --build configure
+  '';
 
-  meta = with lib; {
```

Comment:
why remove mate?

Reply:
Just a draft for concept proof. I'll add it later, apologize!